pontoon/base/models/entity.py

```diff
@@ -879,7 +874,16 @@ def for_project_locale(
                     )
                 )
                 & Q(translation__locale=locale)
-                & q_rejected
+                & (
+                    Q()
+                    if search_rejected_translations
+                    else Q(translation__rejected=False)
+                )
+                | (
+                    Q(**{f"key__{i}regex": rf"{y}{escape(s)}{y}"})
+                    if search_identifiers
+                    else Q()
+                )
```

Comment:
I'm curious why did you include this in `translation_filters` rather than `entity_filters`?

Reply:
I thought that all `Translations` also have a `key` field, but upon checking the admin panel, it seems I was wrong!

I think the reason why this works is because if we leave out the `search_identifiers` query from `translations_filters`, and both `search_identifiers` and `search_translations_only` are set to true, then `entity_filters` is never reached, so the key lookup fails.

Would it be better to instead remove the `if not search_translations_only` statement on line 895 and refactor accordingly?

Reply:
Removing that check would be preferable... Unless it's a massive refactor, in which case I'd just leave it as it is and add a comment. It's possible Eemeli will need to change this code quite a lot as part of the data model unification.